appium_helpers.py

```diff
@@ -0,0 +1,160 @@
+                           s,
+                           wait_time)
+
+    @staticmethod
+    def find_all_webview_elements_by_xpath(driver):
+        return driver.find_elements_by_xpath('//UIAApplication[1]/UIAWindow[1]/UIAScrollView[1]/UIAWebView[1]/*')
+
+    @staticmethod
+    def find_element_by_xpath(driver, xpath):
+        return driver.find_element_by_xpath(xpath)
+
+    @staticmethod
+    def find_webview_elements_by_xpath(driver, xpath):
+        return driver.find_elements_by_xpath('//UIAApplication[1]/UIAWindow[1]/UIAScrollView[1]/UIAWebView[1]/' + xpath)
+
+    @staticmethod
+    def find_all_elements_by_xpath(driver):
+        return driver.find_elements_by_xpath('//*[not(*)]')
+
+    @staticmethod
+    def find_all_window_elements_by_xpath(driver):
+        return driver.find_elements_by_xpath('//UIAApplication[1]/UIAWindow[1]/*')
+
+    @staticmethod
+    def find_all_window_elements(driver):
+        return driver.find_elements_by_ios_uiautomation('.elements()')
+
+    @staticmethod
+    def wait_by(driver, expected_condition, by_method, element_identifier, wait_time=5):
+        """
+        See: http://selenium-python.readthedocs.io/waits.html
+        Args:
+            driver: instance of Appium.webdriver
+            expected_condition: what we condition we expect to happen
+            by_method: by with what Appium.webdriver.method we try to find the element
+            element_identifier: depending on 'by_method' what indentifier we try to use to locate the elem
+            wait_time: time to wait element to appear
+
+        Returns: instance of appium.webdriver.webelement or TimeoutException
+        """
+        return WebDriverWait(driver, wait_time).until(expected_condition((by_method, element_identifier)))
+
+
+
+    @staticmethod
+    def toggle_orientation(driver):
+        current_orientation = driver.orientation
+        if current_orientation == "PORTRAIT":
+            driver.orientation = "LANDSCAPE"
+        else:
+            driver.orientation = "PORTRAIT"
+
+    @staticmethod
+    def get_xml_tree(driver):
+        return driver.execute_script('au.mainApp().getTreeForXML()')
+
+    @staticmethod
+    def screenshot_fast(driver, name, dir=getcwd()+'/screenshots'):
+        """
+        Takes PNG screenshot using command idevicescreenshot.
+        """
+
+        #TODO: we need to define the device we launch the appium-server against in ENV
+        #      This will fail if multiple devices are present
+        udid = check_output(["idevice_id", "-l"]).decode("utf-8").strip('\n')
+        name = str(name) + '.png'
+
+        try:
+            out = check_output(["idevicescreenshot", "-u", str(udid), dir+'/'+name])
+        except CalledProcessError as e:
+            print("Error taking screenshot {} in {} with error {}".format(name, dir, e.output))
+            return False
+        return True
+
+    @staticmethod
+    def screenshot(driver, name, dir=getcwd()+'/screenshots'):
```

Comment:
should this be in `PlatformBase`-class? it looks like it would be same for Android.